ui/src/app/services/error-handling.service.ts

```diff
@@ -0,0 +1,59 @@
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+
+@Injectable()
+export class ErrorHandlingService {
```

Comment:
Have you tried reporting errors from two different components? I'm wondering if Angular knows to create this as a singleton. (There were two UI components + ErrorHandlerComponent, and any one of them instantiated a different ErroHandlingService, things would obviously get confusing.)

Reply:
I have not. I can work on doing that so that two publish at the same time. That said, I know that if I include the service in one component (`AppComponent` for example) and call the functions there, it sets the values in the same service that is in `ErrorHandlerComponent`, so those are at least the same.